app/models/display_ad.rb

```diff
@@ -54,21 +59,25 @@ def self.for_display(area:, user_signed_in:, organization_id: nil, article_id: n
       permit_adjacent_sponsors: permit_adjacent_sponsors,
     )
 
-    # Business Logic Context:
-    # We are always showing more of the good stuff — but we are also always testing the system to give any a chance to
-    # rise to the top. 1 out of every 8 times we show an ad (12.5%), it is totally random. This gives "not yet
-    # evaluated" stuff a chance to get some engagement and start showing up more. If it doesn't get engagement, it
-    # stays in this area.
-
-    # Ads that get engagement have a higher "success rate", and among this category, we sample from the top 15 that
-    # meet that criteria. Within those 15 top "success rates" likely to be clicked, there is a weighting towards the
-    # top ranked outcome as well, and a steady decline over the next 15 — that's because it's not "Here are the top 15
-    # pick one randomly", it is actually "Let's cut off the query at a random limit between 1 and 15 and sample from
-    # that". So basically the "limit" logic will result in 15 sets, and then we sample randomly from there. The
-    # "first ranked" ad will show up in all 15 sets, where as 15 will only show in 1 of the 15.
-    if rand(8) == 1
+    case rand(99) # output integer from 0-99
+    when RARELY # smallest range, 5%
+      # We are always showing more of the good stuff — but we are also always testing the system to give any a chance to
+      # rise to the top. 5 out of every 100 times we show an ad (5%), it is totally random. This gives "not yet
+      # evaluated" stuff a chance to get some engagement and start showing up more. If it doesn't get engagement, it
+      # stays in this area.
       ads_for_display.sample
-    else
+    when SELDOM # medium range, 30%
+      # Here we sample from only billboards with fewer than 1000 impressions (with a fallback
+      # if there are none of those, causing an extra query, but that shouldn't happen very often).
+      ads_for_display.seldom_seen.sample || ads_for_display.sample
```

Comment:
Yes, this is correct with the intention @jaw6.

Is this the _correct overall decision in the long run_? I think that's unlikely, but this will give us the faster feedback we're looking for on newly published billboards, and we will need to make future adjustments — and/or settle on a long-term solution. 🙂